pkg/services/address/address_lookup.go

```diff
@@ -119,3 +120,84 @@ func FindLocationsByZipCity(appCtx appcontext.AppContext, search string, exclusi
 	}
 	return locationList, nil
 }
+
+type vIntlLocation struct {
+}
+
+func NewVIntlLocation() services.VIntlLocation {
+	return &vIntlLocation{}
+}
+
+func (o vIntlLocation) GetOconusLocations(appCtx appcontext.AppContext, country string, search string, exactMatch bool) (*models.VIntlLocations, error) {
+
+	locationList, err := FindOconusLocations(appCtx, country, search, exactMatch)
+
+	if err != nil {
+		switch err {
+		case sql.ErrNoRows:
+			return &locationList, apperror.NewNotFoundError(uuid.Nil, "Search string: "+search)
+		default:
+			return &locationList, err
+		}
+	}
+
+	return &locationList, nil
+}
+
+// Returns a VIntlLocation array containing all results for the search for the given country
+// This method expects a comma to be entered after the city name has been entered and is used
+// to determine when the principal division needs to be parsed from the search string
+func FindOconusLocations(appCtx appcontext.AppContext, country string, search string, exactMatch bool) (models.VIntlLocations, error) {
+	var locationList []models.VIntlLocation
+	searchSlice := strings.Split(search, ",")
+	city := ""
+	principalDivision := ""
+
+	if len(searchSlice) > 1 {
+		city = strings.TrimSpace(searchSlice[0])
+		principalDivision = strings.TrimSpace(searchSlice[1])
+	} else {
+		city = search
+	}
+
+	var queryCondition string
+	if exactMatch {
+		queryCondition = "="
+	} else {
+		queryCondition = "LIKE"
+	}
+
+	sqlQuery := fmt.Sprintf(`
+	    SELECT vil.city_name, vil.country_prn_dv_nm, vil.icc_id, vil.re_country_prn_division_id
+	    FROM v_intl_locations vil
+	    WHERE upper(vil.country) = upper(?)
+	    AND upper(vil.city_name) %s upper(?)
+	    AND upper(vil.country_prn_dv_nm) %s upper(?)
+	    ORDER BY vil.country_prn_dv_nm`,
+		queryCondition, queryCondition,
+	)
+
+	sqlQuery += ` limit 30`
+	var query *pop.Query
+
+	// we only want to add an extra % to the strings if we are using the LIKE in the query
+	if exactMatch {
+		query = appCtx.DB().RawQuery(sqlQuery, country, city, principalDivision)
+	} else {
+		query = appCtx.DB().RawQuery(sqlQuery, country, fmt.Sprintf("%s%%", city), fmt.Sprintf("%s%%", principalDivision))
+	}
```

Comment:
```suggestion
	} else {
		query = appCtx.DB().RawQuery(sqlQuery, country, fmt.Sprintf("%s%%", city), fmt.Sprintf("%%%s%%", principalDivision))
	}
```

Adding `%%` so that we end up with `%query%` for the principal division seems to resolve most scenarios where the user queries without inverting the principal division name to what we store it as.